agent/app/agent.go

```diff
@@ -579,3 +586,29 @@ func (agent *ecsAgent) startACSSession(
 	seelog.Critical("ACS Session handler should never exit")
 	return exitcodes.ExitError
 }
+
+// validateRequiredVersion validates docker version.
+// Minimum docker version supported is 1.9.0, maps to api version 1.21
+// see https://docs.docker.com/develop/sdk/#api-version-matrix
+func (agent *ecsAgent) validateRequiredVersion() (int, bool) {
+	supportedVersions := agent.dockerClient.SupportedVersions()
+	if len(supportedVersions) == 0 {
+		seelog.Critical("Could not get supported docker versions.")
+		return exitcodes.ExitError, false
+	}
+
+	// if api version 1.21 is supported, it means docker version is at least 1.9.0
+	minVersionMet := false
+	for _, version := range supportedVersions {
+		if version == dockerclient.Version_1_21 {
+			minVersionMet = true
```

Comment:
if you `return -1, true` here, you can get rid of this variable and the next if statement. :)